intuita-webview/src/main/Tree/index.tsx

```diff
@@ -1,41 +1,69 @@
 import ReactTreeView from 'react-treeview';
 import { ReactNode, memo, useState } from 'react';
 import { TreeNode } from '../../../../src/components/webview/webviewEvents';
+import { generateRandomColor } from '../../utilities';
 
 type Props = {
+	index: number;
 	depth: number;
 	node: TreeNode;
 	renderItem({
+		index,
 		node,
 		depth,
 		open,
 		setIsOpen,
+		color,
+		isLastChild,
 	}: {
+		index: number;
 		node: TreeNode;
 		depth: number;
 		open: boolean;
 		setIsOpen: (value: boolean) => void;
+		color: string;
+		isLastChild: boolean;
 	}): ReactNode;
+	color: string;
+	isLastChild: boolean;
 };
 
-const Tree = ({ node, depth = 0, renderItem }: Props) => {
+const Tree = ({
+	node,
+	depth = 0,
+	renderItem,
+	color: colorProp,
+	index,
+	isLastChild,
+}: Props) => {
+	const hasNoChildren = !node.children || node.children.length === 0;
+	const isFolderBreakdown = [
+		'folderElement',
+		'acceptedFolderElement',
+	].includes(node.kind);
 	const [open, setIsOpen] = useState(depth === 0);
+	const [color] = useState(hasNoChildren ? colorProp : generateRandomColor());
```

Comment:
as I mentioned in `utilities`, using randomness makes the experience not reproducible

Reply:
addressed